mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp

```diff
@@ -313,7 +316,7 @@ class DotConverter final : public OpConversionPattern<migraphx::DotOp> {
     // A, B, Out have the same rank. rank=2 assumes batch=1.
     // Here handling special cases.
     if (outRank != 3 || rankA != rankB ||
-        (outRank == 3 && orgDimsA != orgDimsB)) {
+        (outRank == 3 && orgDimsA[0] != orgDimsB[0])) {
```

Comment:
Why this change ?

Reply:
This was a bug before. `orgDimsA` will always be different than `orgDimsB`, because the layout is [m, k] * [k, n]. So the effect of the bug is that we'd almost always generate a pair of redundant reshape before and after the dot op.

I think the original intent was to allow the batch dimension to broadcast when they don't align. So I did exactly that check in here that make the redundant reshape to go away.

Reply:
I see... 
Can you add a test for this ? maybe in mixr_to_tosa ?

Reply:
This branch will realistically never happen because migraphx won't implicitly broadcast, but rather always use a multibroadcast op to achieve this. So when it comes to this op, the first dimension must be same between A and B. The original intention to support such kind of broadcast is kind of mis-leading.

But sure, I can add a test to make sure that unrealistic use case do not yield a bug.

Reply:
Done in 17dc34ef9b0913693dd54c4144a84e4e3dd462a3